src/main/java/org/jabref/logic/texparser/DefaultTexParser.java

```diff
@@ -0,0 +1,141 @@
+
+    public DefaultTexParser() {
+        this.result = new TexParserResult();
+    }
+
+    @Override
+    public TexParserResult parse(String citeString) {
+        matchCitation(Paths.get("foo/bar"), 1, citeString);
+        return result;
+    }
+
+    @Override
+    public TexParserResult parse(Path texFile) {
+        return parse(Collections.singletonList(texFile));
+    }
+
+    @Override
+    public TexParserResult parse(List<Path> texFiles) {
+        List<Path> referencedFiles = new ArrayList<>();
+
+        if (result.getFileList().isEmpty()) {
+            result.getFileList().addAll(texFiles);
+        } else {
+            result.getNestedFiles().addAll(texFiles);
+        }
+
+        for (int fileIndex = 0; fileIndex < texFiles.size(); fileIndex++) {
+            Path file = texFiles.get(fileIndex);
+
+            try (LineNumberReader lnr = new LineNumberReader(Files.newBufferedReader(file))) {
+                for (String line = lnr.readLine(); line != null; line = lnr.readLine()) {
+                    if (line.startsWith("%")) {
+                        // Skip comment lines.
+                        continue;
+                    }
+
+                    matchCitation(file, lnr.getLineNumber(), line);
+                    matchNestedFile(file, texFiles, referencedFiles, line);
+                }
+            } catch (IOException e) {
+                LOGGER.warn("Error opening the TEX file", e);
+            }
+        }
+
+        // Parse all files referenced by TEX files, recursively.
+        if (!referencedFiles.isEmpty()) {
+            parse(referencedFiles);
+        }
+
+        return result;
+    }
+
+    /**
+     * Find cites along a specific line and add them to a map.
+     */
+    private void matchCitation(Path file, int lineNumber, String line) {
+        Matcher citeMatch = Pattern.compile(CITE_REGEX).matcher(line);
+
+        while (citeMatch.find()) {
+            String[] keys = citeMatch.group("key").split(",");
+
+            for (String key : keys) {
+                Citation citation = new Citation(file, lineNumber, citeMatch.start(), citeMatch.end(), line);
+
+                if (!result.getCitations().containsKey(key)) {
+                    result.getCitations().put(key, new ArrayList<>());
+                }
+
+                if (!result.getCitations().get(key).contains(citation)) {
+                    result.getCitations().get(key).add(citation);
+                }
+            }
+        }
+    }
+
+    /**
+     * Find inputs and includes along a specific line and store them for parsing later.
+     */
+    private void matchNestedFile(Path file, List<Path> texFiles, List<Path> referencedFiles, String line) {
+        Matcher includeMatch = Pattern.compile(INCLUDE_REGEX).matcher(line);
+
+        while (includeMatch.find()) {
+            String include = includeMatch.group("file");
+
+            if (!include.endsWith(".tex")) {
+                include += ".tex";
```

Comment:
please move the ".tex" to a constant and also the "file" so it's easier to identify and modify.